elf_reader.go

```diff
@@ -575,6 +582,26 @@ func (ec *elfCode) relocateInstruction(ins *asm.Instruction, rel elf.Symbol) err
 			return fmt.Errorf("asm relocation: %s: unsupported type %s", name, typ)
 		}
 
+		// If we arrive here, it can possibly be because we are dealing with
+		// .kconfig.
+		if mapSpec, ok := (*ec.maps)[".kconfig"]; ok {
+			if ds, ok := mapSpec.Value.(*btf.Datasec); ok {
+				for _, varSecInfo := range ds.Vars {
+					variable, ok := varSecInfo.Type.(*btf.Var)
```

Comment:
You already perform this check in `populateDataSections()`. `v := vsi.Type.(*btf.Var)` is fine here.